bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/handler/ViessmannBridgeHandler.java

```diff
@@ -0,0 +1,337 @@
+        } catch (JsonSyntaxException | IllegalStateException e) {
+            logger.warn("Parsing Viessmann response fails: {}", e);
+        }
+    }
+
+    private void startViessmannBridgePolling(Integer pollingIntervalS) {
+        ScheduledFuture<?> currentPollingJob = viessmannBridgePollingJob;
+        if (currentPollingJob == null) {
+            viessmannBridgePollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                logger.debug("Refresh job scheduled to run every {} seconds for '{}'", pollingIntervalS,
+                        getThing().getUID());
+                api.checkExpiringToken();
+                checkResetApiCalls();
+                pollingFeatures();
+            }, 1, TimeUnit.SECONDS.toSeconds(pollingIntervalS), TimeUnit.SECONDS);
+        }
+    }
+
+    private void startViessmannBridgeLimitReset(Long delay) {
+        ScheduledFuture<?> currentPollingJob = viessmannBridgeLimitJob;
+        if (currentPollingJob == null) {
+            viessmannBridgeLimitJob = scheduler.scheduleWithFixedDelay(() -> {
+                logger.debug("Resetting Limit and reconnect for '{}'", getThing().getUID());
+                api.checkExpiringToken();
+                checkResetApiCalls();
+                getAllDevices();
+                if (!devicesList.isEmpty()) {
+                    updateBridgeStatus(ThingStatus.ONLINE);
+                    startViessmannBridgePolling(getPollingInterval());
+                    stopViessmannBridgeLimitReset();
+                }
+            }, delay, TimeUnit.SECONDS.toSeconds(120), TimeUnit.SECONDS);
+        }
+    }
+
+    public void stopViessmannBridgePolling() {
+        ScheduledFuture<?> currentPollingJob = viessmannBridgePollingJob;
+        if (currentPollingJob != null) {
+            currentPollingJob.cancel(true);
+            viessmannBridgePollingJob = null;
+        }
+    }
+
+    public void stopViessmannBridgeLimitReset() {
+        ScheduledFuture<?> currentPollingJob = viessmannBridgeLimitJob;
+        if (currentPollingJob != null) {
+            currentPollingJob.cancel(true);
+            viessmannBridgeLimitJob = null;
+        }
+    }
+
+    public void waitForApiCallLimitReset(Long resetLimitMillis) {
+        stopViessmannBridgePolling();
+        Long delay = (resetLimitMillis - Instant.now().toEpochMilli()) / 1000;
+        stopViessmannBridgeLimitReset();
+        startViessmannBridgeLimitReset(delay);
+    }
+
+    /**
+     * Notify appropriate child thing handlers of an Viessmann message by calling their handleUpdate() methods.
+     *
+     * @param msg message to forward to child handler(s)
+     */
+    private void notifyChildHandlers(FeatureDataDTO msg) {
+        for (Thing thing : getThing().getThings()) {
+            ViessmannThingHandler handler = (ViessmannThingHandler) thing.getHandler();
+            //@formatter:off
+            if (handler != null && (handler instanceof DeviceHandler && msg instanceof FeatureDataDTO)) {
```

Comment:
If `handler instanceof DeviceHandler` is true, it is not null. You can remove the null-check